src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -828,9 +824,38 @@ BsonDocument getResumeToken(final MongoSourceConfig sourceConfig) {
   @Override
   public void commitRecord(final SourceRecord record, final RecordMetadata metadata) {
     if (metadata == null) {
-      currentStatistics.filteredRecords(1);
+      currentStatistics.getRecordsFiltered().sample(1);
     } else {
-      currentStatistics.successfulRecords(1);
+      currentStatistics.getRecordsAcknowledged().sample(1);
+    }
+  }
+
+  private void mongoCommandSucceeded(final CommandSucceededEvent event) {
+    String commandName = event.getCommandName();
+    long elapsedTimeMs = event.getElapsedTime(TimeUnit.MILLISECONDS);
+    if ("getMore".equals(commandName)) {
+      currentStatistics.getSuccessfulGetMoreCommands().sample(elapsedTimeMs);
+    } else if ("aggregate".equals(commandName) || "find".equals(commandName)) {
+      currentStatistics.getSuccessfulInitiatingCommands().sample(elapsedTimeMs);
+    }
+    ResumeTokenUtils.getResponseOffsetSecs(event.getResponse())
+        .ifPresent(offset -> currentStatistics.getLatestOffsetSecs().sample(offset));
+  }
+
+  private void mongoCommandFailed(final CommandFailedEvent event) {
+    Throwable e = event.getThrowable();
+    if (e instanceof MongoCommandException) {
+      if (doesNotSupportsStartAfter((MongoCommandException) e)) {
+        // silently ignore this expected exception, which is used to set this.supportsStartAfter
+        return;
```

Comment:
It could be misleading to ignore these events entirely.  Did you consider whether to treat them as successful commands instead?

Reply:
It seems that ideally, we might hit the server during initialization to see if it supports `startsAfter`. That wouldn't be as efficient. But if we did that, I would expect such a request to not be counted as a normal message, so on that basis I don't count it here. I agree that this might be confusing for users who enable server logging and are trying to match up message counts. But it seems more confusing to log this as a success when it would show up as an error in such logs. And counting it as an error when it is expected might needlessly worry users. Ignoring seems like the least-bad option, though I don't feel particularly strongly about this.